### PR TITLE
Rollout listing

### PR DESCRIPTION
**Pull request by adithyadinesh0412**

### CODE REVIEW

File: src/database/queries/rollouts.js [hunk 1]
```diff
@@ -47,6 +47,7 @@ exports.findAll = async (filter, attributes = {}) => {
 		const res = await Rollout.findAll({
 			where: filter,
 			attributes,
+			...options,
```
Comment (priyanka-TL) [inline src/database/queries/rollouts.js]: @adithyadinesh0412  options is not passing in the params

File: src/database/queries/rollouts.js [hunk 2]
```diff
@@ -55,6 +56,20 @@ exports.findAll = async (filter, attributes = {}) => {
 		return error
 	}
 }
+exports.findAllAndCount = async (filter, attributes = [], options = {}) => {
```
Comment (priyanka-TL) [inline src/database/queries/rollouts.js:58]: @adithyadinesh0412  options is not using here

File: src/services/rollouts.js [hunk 3]
```diff
@@ -122,4 +125,144 @@ module.exports = class RolloutsHelper {
...
+			}
+
+			if (searchText && searchText != '') {
+				filters.title = {
+					[Op.iLike]: '%' + searchText + '%',
+				}
+			}
+
+			if (queryParams.type && queryParams.type != '') {
+				filters.type = queryParams.type
```
Comment (priyanka-TL) [inline src/services/rollouts.js]: @adithyadinesh0412  in resource_type we have the rollout will be project or observation. u need to check with resource_type

File: src/services/rollouts.js [hunk 4]
```diff
@@ -122,4 +125,144 @@ module.exports = class RolloutsHelper {
...
+			}
+
+			if (queryParams.status && queryParams.status != '') {
```
Comment (priyanka-TL) [inline src/services/rollouts.js:161]: @adithyadinesh0412  type will be program or solution, solution u dont need to list . Remove solution type rollout and list

File: dev-ops/integration_test.self_creation_portal.env [hunk 5]
```diff
@@ -82,4 +82,11 @@ DEFAULT_ORGANISATION_CODE=default_code
 ENABLE_OBSERVATION_IN_PROJECTS=true
 
 # Regular expression to validate observation deep link
-OBSERVATION_DEEP_LINK_REGEX= "^https:\/\/dev\.elevate-ml\.shikshalokam\.org\/view\/observation\/[a-f0-9]{32}$"
\ No newline at end of file
+OBSERVATION_DEEP_LINK_REGEX= "^https:\/\/dev\.elevate-ml\.shikshalokam\.org\/view\/observation\/[a-f0-9]{32}$"
+
+#Role name which have permission to rollout the resourced
```
Comment (priyanka-TL) [inline dev-ops/integration_test.self_creation_portal.env:87]: @adithyadinesh0412  update postman collection also

File: src/services/rollouts.js [hunk 6]
```diff
@@ -122,4 +125,144 @@ module.exports = class RolloutsHelper {
...
 		}
 	}
+	/**
+	 * Rollout List
+	 * @method
+	 * @name list
+	 * @param {Object} req - request data.
+	 * @returns {JSON} - List of rollouts
+	 */
+	static async list(organization_id, loggedInUserId, queryParams, searchText = '', page, limit) {
```
Comment (priyanka-TL) [inline src/services/rollouts.js:139]: @adithyadinesh0412  add all parameters in the function signature

File: src/services/rollouts.js [hunk 7]
```diff
@@ -122,4 +125,144 @@ module.exports = class RolloutsHelper {
...
+
+			if (queryParams.status && queryParams.status != '') {
+				filters.status = {
+					[Op.in]: queryParams.status.trim().split(','),
+				}
+			}
+
+			const sort = await this.constructSortOptions(queryParams?.sort_by, queryParams?.sort_order)
+
+			const rolloutList = await rolloutQueries.findAllAndCount(
```
Comment (priyanka-TL) [inline src/services/rollouts.js:169]: @adithyadinesh0412  do we need to show the status as inactive in the listing ?

**Review by priyanka-TL — CHANGES_REQUESTED** [review]
Reviewed on Dec 3, 9.26 PM

File: src/services/rollouts.js (hunk 7 above)
Comment (adithyadinesh0412) [reply src/services/rollouts.js:169]: That was what was concluded right? @priyanka-TL . To show the Inactive rollouts. please correct me if im wrong.

**Review by priyanka-TL — APPROVED** [review]
Approved on Dec 3rd

**Pull request merged**